### PR TITLE
[#74] Add proper README and project description

### LINKED ISSUE #74: Add Readme and project description
# Clarification and motivation

We should fill in the project's `README`.
This should include:
* A short description of what coffer does + will do in the future.
* Which problems with the existing state of the art we're trying to address.
* How to install, configure, and use the tool
* How to setup a local dev environment
  * This includes how to setup a local vault instance
  * How to run unit & golden tests

We should also:
* write a short description in the `description` field of `package.yaml`.
* add a description to the CLI's `--help` command (see `TODO`s in `CLI.Parser.parserInfo`)

# Acceptance criteria

The following have been updated: `README.md`, `package.yaml` and `CLI.Parser.parserInfo`.

### PR DESCRIPTION
**Pull request by MagicRB**
Signed-off-by: Magic_RB <magic_rb@redalder.org>

## Description

Update `README.md`, add description to `package.yaml` and `CLI.Parser.parserInfo`

## Related issue(s)

#62, #56, #37, we need at least one of these done, so that we can start writing a proper `/doc/`, without another backend, it's hard to write good docs.

It would also have a stable release, we could point to in the README, so this is blocked by an MVP too.

Fixed #74 

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [X] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [X] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [X] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by MagicRB** [comment]
The logo does't have a license file as of now, idk what to license it under, some CC probably would be good.

File: README.md [hunk 1]
```diff
@@ -4,4 +4,96 @@
...
+name = "vault-local"
+address = "localhost:8200"
+mount = "secret"
+token = "<vault token>"
+
+```
+
+The configuration consists of a single `main_backend` key and N `[[backend]]` sections. Each one must at least have the `type` key, specify which backend you wish to use. The rest depends on each backend, to see a comprehensive list, refer to *backends*.
+
+The coffer CLI will look for it's configuration file, in the current working directory for now.
```
[the remark this answers is not in the document]
Comment (dcastro) [reply README.md:37]: This isn't exactly right.

The path to the config file can be specified via the `-c/--config` command-line option or via the `COFFER_CONFIG` env variable. If none are used, it'll look for a file named `config.toml` in the current working directory.

File: lib/CLI/Parser.hs [hunk 2]
```diff
@@ -44,8 +44,11 @@ parserInfo :: ParserInfo Options
 parserInfo =
   info (parser <**> helper) $
     fullDesc
-    <> progDesc "TODO: coffer description goes here"
-    <> header "TODO: coffer description goes here"
+    <> progDesc (unlines [ "Coffer CLI is just one frontend of the coffer project."
+                         , "Coffer, the project, is a multi-backend, multi-frontend password manager."
+                         , "It reaches out to multiple backends, allowing you to manage multiple password stores af it were one. From anywhere!"
+                         ])
```
Comment (dcastro) [inline lib/CLI/Parser.hs:50]: :+1:

File: README.md [hunk 3]
```diff
@@ -4,4 +4,96 @@
  - SPDX-License-Identifier: MPL-2.0
  -->
 
-# coffer
+# Coffer
+
+![coffer-logo](./logo.png "Coffer logo")
+
+Coffer is multi-backend password store with multiple frontends. It reaches out to Vault, pass and others to store your most precious secrets.
```
Comment (dcastro) [inline README.md:11]:
```suggestion
Coffer is multi-backend password store with multiple frontends. It reaches out to [Vault](https://www.vaultproject.io/), [pass](https://www.passwordstore.org/) and others to store your most precious secrets.
```

**Review by dcastro — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -4,4 +4,96 @@
...
+```
+
+## Development
+
+### Nix
+
+This is the preffered way, install Nix according to the [Nix download page](https://nixos.org/download.html "nix download"). Then execute:
+
+```shell
+$ nix --exprimental-features 'nix-command flakes' shell .#
```
[the remark this answers is not in the document]
Comment (MagicRB) [reply README.md:72]: We can add stack to `flake.nix`under devShell nativeBuildInputs

[the remark this answers is not in the document]
Comment (MagicRB) [reply README.md:72]: The fact that Vault is needed should be specified in sperate Vault backend specific file

Comment (dcastro) [inline README.md:72]:
```
nix develop --extra-experimental-features 'nix-command flakes'
```

[the remark this answers is not in the document]
Comment (dcastro) [reply README.md:72]: And link it from `README.md`? Sure.

Just wanna make sure that any new dev joining the team can just follow the readme and get everything up and running on their own

File: README.md [hunk 5]
```diff
@@ -4,4 +4,96 @@
...
+> entry_path - path to an entry, think of it as a path to a file instead of
+			   a directory
+> path - path to a directory, not an entry
+> field_key - the key of a field, must satisfy this regex: `?regex?`
+> field_value - the value of a field, must satisfy this regex: `?regex?`
+> tag - a tag, assigned to an entry to tag it, must satisfy this regex:
+		`?regex?`
+```
+
+## Development
```
Comment (dcastro) [inline README.md:65]: We should have a sub-section for testing under `Development`.

From the description in #74:
> How to run unit & golden tests

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (DK318) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (DK318) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (DK318) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (DK318) [reply README.md:68]: Done

File: README.md [hunk 6]
```diff
@@ -4,4 +4,217 @@
...
+> path - path to a directory, not an entry
+> field_key - the key of a field, must satisfy this regex: `?regex?`
+> field_value - the value of a field, must satisfy this regex: `?regex?`
+> tag - a tag, assigned to an entry to tag it, must satisfy this regex:
+		`?regex?`
+```
+
+### Web API
+
+Coffer also have a Web API. Currently it runs on `localhost:8081`. In the future we could configure this but at this moment we can't.
```
Comment (dcastro) [inline README.md:72]:
```suggestion
Coffer also has a Web API. Currently, it runs on `localhost:8081`. In the future we could configure this but at this moment we can't.
```

File: README.md [hunk 7]
```diff
@@ -4,4 +4,217 @@
...
+
```
Comment (dcastro) [inline README.md:73]: Let's also explain how to start the server, i.e. with `stack run coffer-server` or `cabal run coffer-server`

**Review by dcastro — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dcastro) [reply README.md]: @DK318 Can you do this bit as well? Either add a new section for describing the backends or a separate file.

[the remark this answers is not in the document]
Comment (dcastro) [reply README.md]: @DK318 can you do this bit as well please?

[the remark this answers is not in the document]
Comment (DK318) [reply README.md]: Done

[the remark this answers is not in the document]
Comment (DK318) [reply README.md]: Done

File: README.md [hunk 8]
```diff
@@ -4,4 +4,217 @@
...
+Coffer is multi-backend password store with multiple frontends. It reaches out to [Vault](https://www.vaultproject.io/), [pass](https://www.passwordstore.org/) and others to store your most precious secrets.
+
+## Installation
+
+??? from releases?
+
+## Usage
+
+### Configuration
```
[the remark this answers is not in the document]
Comment (DK318) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (DK318) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (DK318) [reply README.md]: Done

File: docs/cli-backends-configuration.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+### Backend configuration file example
+```toml
+type = "vault-kv" # type for 'vault' backend
+address = "localhost:8200" # required
+mount = "secret" # required
+token = "<vault token>" # required
+```
+
+### How to run
+First of all, you need to install `vault` from it's [official site](https://www.vaultproject.io/). Then you can start server with the next command:
```
Comment (dcastro) [inline docs/cli-backends-configuration.md:20]:
```suggestion
First of all, you need to install `vault` from its [official site](https://www.vaultproject.io/). Then you can start server with the next command:
```

**Review by dcastro — APPROVED** [review]
LGTM :+1:

**Pull request merged**